Source/Core/DolphinQt/NetPlay/NetPlayDialog.cpp

```diff
@@ -461,6 +462,55 @@ void NetPlayDialog::show(std::string nickname, bool use_traversal)
   UpdateGUI();
 }
 
+void NetPlayDialog::UpdateDiscordPresence()
+{
+#ifdef USE_DISCORD_PRESENCE
+  // both m_current_game and m_player_count need to be set for the status to be displayed correctly
+  if (m_player_count == 0 || m_current_game.empty())
+    return;
+
+  const auto use_default = [this]() {
+    Discord::UpdateDiscordPresence(m_player_count, Discord::SecretType::Empty, "", m_current_game);
+  };
+
+  if (g_TraversalClient)
+  {
+    const auto host_id = g_TraversalClient->GetHostID();
+    if (host_id == decltype(host_id)())
+      return use_default();
+
+    Discord::UpdateDiscordPresence(m_player_count, Discord::SecretType::RoomID,
+                                   std::string(host_id.begin(), host_id.end()), m_current_game);
+  }
+  else if (IsHosting())
+  {
+    if (m_exernal_ip_address.empty())
+    {
+      Common::HttpRequest request;
+      Common::HttpRequest::Response response =
+          request.Get("https://ip.dolphin-emu.org/", {{"X-Is-Dolphin", "1"}});
```

Comment:
We need to force IPv4 resolution here, as ENet does not support IPv6. The best way would be to add a function in `Common::HttpRequest` for setting `CURLOPT_IPRESOLVE` to `CURL_IPRESOLVE_V4`.